backend/review/views.py

```diff
@@ -666,17 +748,26 @@ def autocomplete(request):
     to improve performance.
     """
     courses = (
-        Course.objects.filter(course_filters_pcr_allow_xlist)
+        Course.objects.filter(course_filters_pcr)
         .order_by("semester")
-        .values("full_code", "title")
+        .annotate(
+            max_semester=Subquery(
+                Course.objects.filter(full_code=OuterRef("full_code"), title=OuterRef("title"))
+                .annotate(common=Value(1))
+                .values("common")
+                .annotate(max_semester=Max("semester"))
+                .values("max_semester")
+            )
+        )
+        .values("full_code", "title", "max_semester")
         .distinct()
     )
     course_set = sorted(
         [
             {
                 "title": course["full_code"],
                 "desc": [course["title"]],
-                "url": f"/course/{course['full_code']}",
+                "url": f"/course/{course['full_code']}/{course['max_semester']}",
```

Comment:
Why is the semester appended to the end of the URL here?

Reply:
I think it's fine to fully qualify the URL in links clicked by search results - then anyone who copies the link from the URL will have a permalink to that specific course for that topic, rather than future superceding courses. The returned page will be the same as long as the course hasn't been superceded.